src/main/java/com/amazonaws/services/sqs/AmazonSQSVirtualQueuesClient.java

```diff
@@ -235,14 +302,22 @@ public HostQueue(String queueUrl) {
             // wait time on the queue.
             this.buffer = new ReceiveQueueBuffer(amazonSqsToBeExtended, queueUrl);
             this.consumer = new SQSMessageConsumer(AmazonSQSVirtualQueuesClient.this, queueUrl, this::dispatchMessage);
+			this.consumer.setConcurrency(AmazonSQSVirtualQueuesClient.this.concurrentyToPollingQueue);
+            this.consumer.setWaitTimeSeconds(AmazonSQSVirtualQueuesClient.this.waitTimeSeconds);
             this.consumer.start();
         }
 
         private void dispatchMessage(Message message) {
             String queueName = message.getMessageAttributes().get(VIRTUAL_QUEUE_NAME_ATTRIBUTE).getStringValue();
             VirtualQueue virtualQueue = virtualQueues.get(queueName);
             if (virtualQueue != null) {
-                virtualQueue.receiveBuffer.deliverMessages(Collections.singletonList(message), queueUrl, null);
+				
+				if ( virtualQueue.messageHandler != null ){
+					virtualQueue.messageHandler.handleMessage(message);
```

Comment:
I'm not really a fan of this feature - besides complicating the API, it means you're executing client-provided callbacks on the dispatching thread, which means you're potentially blocking dispatching messages to other virtual queues while processing this message.

Reply:
We need this because, if we poll virtual queue then it creates one live thread for each virtual queue. Due to more threads, it causing problem on performance.

Reply:
As discussed offline, I'm okay with adding this as long as the dangers are well-documented.